lab-shelly/lib/storage.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const debug = require('debug')('http:storage');
+const storage = {};
+
+module.exports = exports = {};
+
+exports.createItem = function(schema, item) {
+  debug('#createItem');
+    //schema is the data model (cat). item is new cat object which has id, name, and mood.
+  if(!schema) return Promise.reject(new Error('schema required'));
+  if(!item) return Promise.reject(new Error('item required'));
+  if(!storage[schema]) storage[schema] = {};
+//if not set to storage yet, then set a property on the storage object, defined as the schema
+  storage[schema][item.id] = item;
+//storage.cat.id = item we passed in
+  return Promise.resolve(item);
+};
+
+exports.fetchItem = function(schema, id) {
+  debug('#fetchItem');
+
+  return new Promise((resolve, reject) => {
+    if(!schema) return reject(new Error('schema required'));
+    if(!id) return reject(new Error('id required'));
+//go look for storage.cat
+    let schemaName = storage[schema];
+    if(!schemaName) return reject(new Error('schema not found'));
+//look for cat data object with matching id and if found, set it to item
```

Comment:
While I think its a great idea to have comments that help describe what a block is doing, try to manage these comments within a separate branch than the one you are submitting. Generally, its just best practice to not submit code with a ton of comments. So I do have to dock 1 point for them.